### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by tosy7005** [Illumina/interop#165]
I got this error message when I builded the source code.

In file included from
/gpfs1/home/htae/tar_balls/interop/src/tests/interop/metrics/metric_streams_test.cpp:14:
/gpfs1/home/htae/tar_balls/build/external/gtest/src/gtest/include/gtest/gtest.h: In function ‘testing::AssertionResult
testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = illumina::interop::constants::metric_group, T2 = long unsigned int]’:
/gpfs1/home/htae/tar_balls/build/external/gtest/src/gtest/include/gtest/gtest.h:1485:   instantiated from ‘static testing::AssertionResult
testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*,const char*, const T1&, const T2&) [with T1 =illumina::interop::constants::metric_group, T2 = size_t, bool lhs_is_null_literal = false]’
/gpfs1/home/htae/tar_balls/interop/src/tests/interop/metrics/metric_streams_test.cpp:136:   instantiated from here
/gpfs1/home/htae/tar_balls/build/external/gtest/src/gtest/include/gtest/gtest.h:1448: error: comparison between signed and unsigned integerexpressions
gmake[2]: ***
[src/tests/interop/CMakeFiles/interop_gtests.dir/metrics/metric_streams_test.cpp.o] Error 1
gmake[1]: *** [src/tests/interop/CMakeFiles/interop_gtests.dir/all] Error 2
gmake: *** [all] Error 2

So, I changed
/gpfs1/home/htae/tar_balls/build/external/gtest/src/gtest/include/gtest/gtest.h:1448 to
  if (expected == (const T1)actual)

succeed to compile it, but when I run src/examples/example1, I am keep getting 'Segmentation fault’. Is it because of the change?

I am also failing to run this
index-summary /gpfs1/drop-cmi/NextSeqIn/170604_NB501383_0150_AHTKLYBGX2

Is there anyway to find the reason why I am getting 'Segmentation fault’?

**Comment by ezralanglois** [comment]
Hi,

Thanks for the bug report.

It would help to know the following:

1. What compiler are you using
2. What OS

The change is unrelated to the segfault.

Also, is 170604_NB501383_0150_AHTKLYBGX2 a local run or something from BaseSpace?
If the former, can you upload the interops and RunInfo.xml if the later, can you provide a link?

Best,
Rob

**Comment by ezralanglois** [comment]
Also, what version of InterOp are you using?
If it is not the latest version, then I suggest try the latest version first.

**Comment by tosy7005** [comment]
OS is CentOS release 6.4. I used cmake 3.6.2 to compile the source. And InterOp I've compiled should be the latest version. 170604_NB501383_0150_AHTKLYBGX2 is a local run.

This is the link for the RunInfo.xml
https://drive.google.com/file/d/19cN0QY_Uufxxx2srazm3r1ihqRv0WZYe/view?usp=sharing

**Comment by ezralanglois** [comment]
Thanks for the quick response.

What GCC compiler version are you using?
Also, could you provide the IndexMetricsOut.bin?

I should be able to reproduce the segfault here and fix it.

Best,
Rob

**Comment by tosy7005** [comment]
gcc version is  4.4.7. 

I cannot find IndexMetricsOut.bin. These are files in InterOp/
CorrectedIntMetricsOut.bin, EventMetricsOut.bin, PFGridMetricsOut.bin, RegistrationMetricsOut.bin
EmpiricalPhasingMetricsOut.bin, ExtractionMetricsOut.bin, QMetricsOut.bin, TileMetricsOut.bin

Thank you.

**Comment by ezralanglois** [comment]
If IndexMetricsOut.bin is missing, then bcl2fastq has not been run yet.
Also, index-summary will not give you any output.
That said, it should not segfault so I will try reproduce the issue.

It is possible this is a problem with a the compiler, so you can try the Linux build. I have tested it on CentOS 6.
https://github.com/Illumina/interop/releases/download/v1.1.3/InterOp-1.1.3-Linux-GNU.tar.gz

Best,
Rob

**Comment by ezralanglois** [comment]
Also, it would really help if you could provide TileMetricsOut.bin

Best,
Rob

**Comment by tosy7005** [comment]
Thank you. The compiled package works. 

We run bcl2fastq to write outputs in another location, that is why IndexMetricsOut.bin is not in the sample folder.

This is the link for TileMetricsOut.bin
https://drive.google.com/open?id=1bP5bydhwf1JLMUxMSUQosk4SjSgNyZk7

Thank you.

**Comment by ezralanglois** [comment]
I was unable to reproduce the issue with the latest version of the code. 

What version of InterOp were you using when it segfaulted? Was it latest master?

Best,
Rob

**Comment by tosy7005** [comment]
Yes. It is the latest version. I downloaded it last week.

Thank you.

**Comment by ezralanglois** [comment]
I wanted to check an assumption I made. When you run `index-summary` on the run folder was it the original run folder without IndexMetricsOut.bin?

or was it another run folder that contained this metric file? if so, what other files did it contain?

Best,
Rob

**Comment by tosy7005** [comment]
Because of the limited writing permission for the account running bcl2fastq, we generate FASTQ files and IndexMetricsOut.bin to another location, not in the original flowcell folder.
After I saw your comment about IndexMetricsOut.bin, I copied the file into the InterOp folder and ran the program, but still got the segfalt. 
I guess that example1 and example2, which produced the error, don't need IndexMetricsOut.bin?

 These are files in InterOp/
CorrectedIntMetricsOut.bin, EventMetricsOut.bin, PFGridMetricsOut.bin, RegistrationMetricsOut.bin
EmpiricalPhasingMetricsOut.bin, ExtractionMetricsOut.bin, QMetricsOut.bin, TileMetricsOut.bin

Thank you.
